borderless/explainer.md

```diff
@@ -0,0 +1,512 @@
+    - [WCO extension](#wco-extension)
+      - [Reasonings for discarding](#reasonings-for-discarding)
+    - [New JS APIs](#new-js-apis)
+      - [Reasonings for discarding](#reasonings-for-discarding-1)
+  - [Security Considerations](#security-considerations)
+    - [Spoofing risks](#spoofing-risks)
+    - [Out-of-scope Navigation](#out-of-scope-navigation)
+    - [Opening a popup from a borderless PWA](#opening-a-popup-from-a-borderless-pwa)
+    - [Iframes](#iframes)
+
+## Introduction
+
+[VDI](https://www.softwaretestinghelp.com/best-vdi-software/#What_Is_Virtual_Desktop_Infrastructure)
+(Virtual Desktop Infrastructure) providers attempt to mirror remote applications
+to the local desktop. For example, the VDI provider's cloud is running a Text
+Processor (a desktop application window), which is to be presented on the client
+desktop as if the application was running locally.
+
+Installed web apps define their display mode in the manifest.json using display
+and
+[display_override](https://github.com/WICG/display-override/blob/main/explainer.md)
+fields. Currently
+[supported values for display modes](<https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/public/mojom/manifest/display_mode.mojom?q=-file:(test%7CDebug%7C%5C.xtb$)%20case:yes%20third_party%2Fblink%2Fpublic%2Fmojom%2Fmanifest%2Fdisplay_mode.mojom&ss=chromium>)
+are `window-controls-overlay`, `tabbed`, `minimal-ui`, `standalone`, `browser`
+and `fullscreen`. See
+[Manifest/display_override#values](https://developer.mozilla.org/en-US/docs/Web/Manifest/display_override#values)
+for more. From the before-mentioned options, `window-controls-overlay` has some
+same qualities to what we are trying to achieve. However, it is still not
+offering enough native-like experience for some of the use-cases and thus this
+explainer will explain how the title bar will be totally removed and enabling
+so-called borderless mode where the title bar area would be replaced with web
+content and so giving the developers full control on how the title bar would
+look like.
+
+## Example of a complex title bar with window-controls-overlay
+
+Some highly customized title bars would still look clumsy with
+`window-controls-overlay` enforcing the style of the windowing controls
+(minimize, maximize, close etc.).
+
+![clumsy app with WCO](./images/clumsy-app-with-wco.png)
+
+## Goals
+
+- Enable full control over the appearance of the title bar area
+- Ensure draggable regions work with borderless mode
+
+## Non-goals
+
+- Implementation of
+  [additional windowing controls](https://github.com/ivansandrk/additional-windowing-controls/blob/main/awc-explainer.md),
+  which is another closely-related project
+- Making borderless mode dynamic (with e.g. JavaScript APIs or something else)
+- Changing display mode in app settings (e.g. from borderless to standalone)
+- Support for mobileOSs/macOS/windows
```

Comment:
We shouldn't have a W3C spec that says supporting macOS and Windows are non-goals. This may not be a development priority for the team right now, but with my W3C hat on, we have to think about how this extends to other platforms as well as ChromeOS.

The explainer should explore whether this is at least possible on Windows, macOS and Linux (I believe it is).

(Note: I think it's OK to have a non-goal of supporting mobile, since there's very little distinction between `borderless` and `fullscreen` there.)